core/src/main/java/org/bitcoinj/wallet/Wallet.java

```diff
@@ -5346,29 +5326,55 @@ public void setKeyRotationTime(@Nullable Date time) {
      * The given time cannot be in the future.
      * </p>
      */
-    public void setKeyRotationTime(long unixTimeSeconds) {
-        checkArgument(unixTimeSeconds <= TimeUtils.currentTimeSeconds(), "Given time (%s) cannot be in the future.",
-                TimeUtils.dateTimeFormat(unixTimeSeconds * 1000));
-        vKeyRotationTimestamp = unixTimeSeconds;
+    public void setKeyRotationTime(Instant keyRotationTime) {
+        checkArgument(keyRotationTime.compareTo(TimeUtils.currentTime()) <= 0,
+                "Given time (%s) cannot be in the future.",
+                TimeUtils.dateTimeFormat(keyRotationTime.toEpochMilli()));
+        vKeyRotationTime = keyRotationTime;
         saveNow();
     }
 
+    /** @deprecated use {@link #setKeyRotationTime(Instant)} */
+    @Deprecated
+    public void setKeyRotationTime(long timeSecs) {
+        if (timeSecs == 0)
+            vKeyRotationTime = null;
```

Comment:
I think setKeyRotationTime() should be called here, like this:
```
setKeyRotationTime(null);
```

Or maybe use an intermediate, nullable value and call setKeyRotationTime() once.

Reply:
Done.